src/Pdf.php

```diff
@@ -231,7 +231,7 @@ public function fillForm($data, $encoding = 'UTF-8', $dropXfa = true)
         $this->constrainSingleFile();
         $this->getCommand()
             ->setOperation('fill_form')
-            ->setOperationArgument(is_array($data) ? new FdfFile($data, null, null, null, $encoding) : $data, true);
+            ->setOperationArgument(is_array($data) ? new XfdfFile($data, null, null, null, $encoding) : $data, true);
```

Comment:
Instead of hardcoding the file format here, we should make it configurable. So I suggest to add another argument `$format = 'fdf'` to the `fillForm()` method. Valid values would be `fdf` and `xfdf` with `fdf` being the default. This way we make sure, the library is backwards compatible. That's important because some may already use it and FDF works fine for them and maybe XFDF introduces other problems. I don't want to give people trouble when they update.

I'd prefer a solution like this:

```
if (is_array($data)) {
    $className = $format==='fdf' ? 'FdfFile' : 'XfdfFile';
    $data = new $className($data, null, null, null, $encoding);
}
...
->setOperationArgument($data, true);
```